library/Vanilla/Models/FsThemeProvider.php

```diff
@@ -195,6 +195,31 @@ private function normalizeTheme(Addon $theme, array $assets): array {
             }
         }
 
+
+        // Check theme for default.
+        if (!$logo) {
+            if (valr("assets.variables", $res)) {
+                $themeVars = json_decode($res['assets']['variables']->getData(), true);
+                $desktopLogo = valr("titleBar.logo.desktop.url", $themeVars);
+                $mobileLogo = valr("titleBar.logo.mobile.url", $themeVars);
+                $noDesktopLogo = empty($desktopLogo);
+                $noMobileLogo = empty($mobileLogo);
+
+                if (!$noDesktopLogo) {
+                    $res["assets"]["logo"] = new ImageAsset(asset($desktopLogo, true, true));
+                }
+                if (!$noMobileLogo || !$noDesktopLogo) {
+                    if (!$noMobileLogo) {
+                        $res["assets"]["mobileLogo"] = new ImageAsset(asset($mobileLogo, true, true));
+                    } else {
+                        // Use same logo if mobile is not set.
+                        $res["assets"]["mobileLogo"] = new ImageAsset(asset($desktopLogo, true, true));
```

Comment:
Looking at this again could we move the actual `asset()` calls into the `ImageAsset` class?